ui/core-react/src/core-react/l10n/usePackageTranslation.tsx

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import { useLocalization } from "./LocalizationProvider";
+
+/** Used by AppUI packages internally to define package specific `useTranslation` hook.
+ * Uses a localization provider to translate the provided key with a specified namespace. Multiple fallback layers are used to resolve the translation:
+ * - Return value of a specified `fallback` function.
+ * - Default value looked up from a specified `defaults` object.
+ * - Lastly `key` argument of the translation function is returned.
+ * @internal
+ */
+export function usePackageTranslation({
+  namespace,
+  fallback,
+  defaults,
+}: {
+  namespace: string;
+  fallback: (key: string) => string | undefined;
+  defaults: object;
+}) {
+  const localization = useLocalization();
+  const [registered, setRegistered] = React.useState(false);
+  React.useEffect(() => {
+    if (!localization) return;
+
+    let ignore = false;
```

Comment:
I think `ignore` is not needed in this case. It's ok to call `setState` on unmounted component when it is called only once. The warning about setting state in unmounted component is for cases where set state is called from some king of subscription that is not unsubscribed after unmount: https://github.com/facebook/react/pull/22114

Reply:
Great suggestion, but I think we'll have to wait, since this change is only on React18, while we still support v17.
I've added this as a task item: https://github.com/iTwin/appui/issues/786